File: coding-test/nodejs/src/practice/add-of-matrices.js
function solution(arr1, arr2) {
  const answer = [[]]

  for (let i = 0; i < arr1.length; i++) {
    answer[i] = []

    for (let j = 0; j < arr1[0].length; j++) {
      answer[i][j] = arr1[i][j] + arr2[i][j]
    }
  }

  return answer
}

function diffSolution(arr1, arr2) {
  return arr1.map((arr, i) => arr.map((x, j) => x + arr2[i][j]))
}

const arr1 = [
  [1, 2],
  [2, 3],
]
const arr2 = [
  [3, 4],
  [5, 6],
]
console.log(solution(arr1, arr2))
console.log(diffSolution(arr1, arr2))

File: coding-test/java-kotlin/src/kr/co/programmers/java/practice/AddOfMatrices.java
package kr.co.programmers.java.practice;

import java.util.Arrays;

public class AddOfMatrices {

    public static void main(String[] args) {
        int[][] arr1 = {
                {1, 2},
                {2, 3}
        };
        int[][] arr2 = {
                {3, 4},
                {5, 6}
        };

        int[][] result = new Solution().solution(arr1, arr2);
        Arrays.stream(result).forEach(x -> System.out.println(Arrays.toString(x)));
    }

    static class Solution {
        public int[][] solution(int[][] arr1, int[][] arr2) {
            int[][] answer = new int[arr1.length][arr1[0].length];

            for(int i = 0; i < arr1.length; i++){
                for(int j = 0; j < arr1[i].length; j++){
                    answer[i][j] = arr1[i][j] + arr2[i][j];
                }
            }
            return answer;
        }
    }
}
